content/chapitres/tech-lifecycle.adoc

```diff
@@ -0,0 +1,466 @@
+
+== Exemple Maven : Les phases de Maven
+
+* `clean` - Nettoie le contenu du dossier `./target`
+* `validate` - Validation du projet (syntaxe du `pom.xml` et du Java, etc.)
+* `compile` - Fabrication des fichiers `.class` depuis le code java
+* `test` - Exécuter les tests unitaires
+* `package` - Préparer le livrable finale (`.jar` par exemple)
+* `verify` - Exécuter les tests d'intégration
+* `install` - Mettre à disposition le livrable localement pour d'autres projets Maven
+* `deploy` - Copier le livrable dans un système de stockage de dépendance distant
+
+== Exemple Maven : C'est à vous !
+
+=> C'est à vous :
+
+* Ouvrez link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement de démo dans GitPod,window=_blank]
+* On va nettoyer ce qu'on a déjà produit:
++
+[source,bash]
+----
+mvn clean # Supprime le contenu du dossier `./target
+ls -l ./target
+----
+* ⚠️ La phase `clean` est spéciale et n'est jamais appellée par les autres phases
+
+== Exemple Maven : Compiler
+
+* But: Compilez le code source et visualisez le contenu du dossier `target` :
++
+[source,bash]
+----
+ls -l ./target # Avant
+mvn compile
+ls -l ./target # Après
+----
+
+[.notes]
+--
+* Résolution des dépendances
+* Pré-traitement du code source
+* Compilation des classes
+--
+
+== Exemple Maven : Tests Unitaires
+
+* But: Compilez le code source et visualisez le contenu du dossier `target` :
++
+[source,bash]
+----
+ls -l ./target # Avant
+mvn test
+ls -l ./target # Après
+ls -l ./target/surefire-reports
+----
+
+[.notes]
+--
+* Exécute le goal `compile`
+* Compilation des tests unitaires
+* Exécution des tests unitaires
+* Rapports de tests dans *`./target/surefire-reports`*
+--
+
+== Exemple Maven : Fabriquer le livrable
+
+* But: Générez le livrable au format `.jar` :
++
+[source,bash]
+----
+ls -l ./target # Avant
+mvn package
+ls -l ./target # Après
+----
+
+[.notes]
+--
+* Exécute les goals `compile` et `test`
+* Paquetage de l'application
+--
+
+== Exercice Maven : Tests d'intégration
+
+* But: Exécuter les tests d'intégrations et afficher les rapport de tests
+** ⚠️ Piège : regardez bien le contenu du dossier `./target`
+
+== Solution : Tests d'intégration
+
+[source,bash]
+----
+ls -l ./target # Avant
+mvn verify
+ls -l ./target/failsafe-reports
+----
+
+[.notes]
+--
+* Exécute `compile`, `test` et `package`
+* Compile et exécute les tests d'intégration
+sur l'application empaquetée
+* Rapport de tests dans *`./target/failsafe-reports`*
+--
+
+// == Exemple Maven : Exécution
+
+// * Spring Boot demo exécutée comme un *"Über-Jar"*
+// * Exécution avec la commande java:
+// +
+// [source,subs="attributes",bash]
+// ----
+// java -jar ./target/demoapp.jar
+// # Utiliser CTRL-C pour arrêter
+// ----
+// * Ouvrir une autre instance de *DevBox* (link:{devbox-url}[Click Me, window=_blank]) et vérifier
+// link:http//localhost:8080[] avec `curl`
+// +
+// [source,subs="attributes",bash]
+// ----
+// curl -v http://localhost:8080/
+// ----
```

Comment:
Are you sure you want to comment this content ?

Reply:
Still not sure as it might be out of the area of the chapter, but the fact that you say it might be a way to take a decision. Let's try adding it now.